moveit_core/collision_detection_fcl/include/moveit/collision_detection_fcl/collision_common.h

```diff
@@ -213,47 +236,84 @@ struct FCLGeometry
     collision_geometry_->setUserData(collision_geometry_data_.get());
   }
 
+  /** \brief Pointer to FCL collision geometry. */
   std::shared_ptr<fcl::CollisionGeometryd> collision_geometry_;
+
+  /** \brief Pointer to the user defined geometry data. */
   CollisionGeometryDataPtr collision_geometry_data_;
 };
 
 typedef std::shared_ptr<fcl::CollisionObjectd> FCLCollisionObjectPtr;
 typedef std::shared_ptr<const fcl::CollisionObjectd> FCLCollisionObjectConstPtr;
 
+/** \brief A general hight level object which consists of multiple \e FCLCollisionObjects. It is the top level data
+ *  structure which is used in the collision checking process. */
 struct FCLObject
 {
   void registerTo(fcl::BroadPhaseCollisionManagerd* manager);
   void unregisterFrom(fcl::BroadPhaseCollisionManagerd* manager);
   void clear();
 
   std::vector<FCLCollisionObjectPtr> collision_objects_;
+
+  /** \brief To \e collision_objects_ corresponding geometry data. */
   std::vector<FCLGeometryConstPtr> collision_geometry_;
 };
 
+/** \brief Bundles an \e FCLObject and and a broadphase FCL collision manager. */
 struct FCLManager
 {
   FCLObject object_;
   std::shared_ptr<fcl::BroadPhaseCollisionManagerd> manager_;
 };
 
+/** \brief Callback function used by the FCLManager used for each pair of collision objects to
+*   calculate object distances.
```

Comment:
```suggestion
*   calculate object contact information.
```

This callback doesn't calculate distances, does it?

Reply:
It actually does depending on the [case](https://github.com/ros-planning/moveit/blob/167642b3b5d5484295c93478357f18ef0949027c/moveit_core/collision_detection_fcl/src/collision_common.cpp#L182), but contact information is better as it could also include the distance.